pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -43,28 +46,16 @@
  */
 public class CommonsConfigurationUtils {
   private static final Character DEFAULT_LIST_DELIMITER = ',';
+  public static final String VERSION_HEADER_IDENTIFIER = "version";
```

Comment:
SEGMENT_METADATA_VERSION_HEADER_IDENTIFIER = "version" as we only use versions to decide the reader/writer for the segment metadata, and leaving all other users of this config util intact.

Reply:
I think this can be common though. If it is explicitly specified (in any configuration file), we take it; if it is not explicitly specified, we treat it as default version (maybe adding a constant for default version?)

Reply:
I think we can use this header version in specific other scenarios, like moving away from the usage of `LegacyListDelimiterHandler` in future.

Reply:
To be generic, should we rename the SegmentMetadataIOFactory to something like VersionedIOFactory? so that it's not just for reading/writing SegmentMetadata files.